app/uk/gov/hmrc/tradergoodsprofiles/controllers/GetRecordsController.scala

```diff
@@ -48,12 +49,26 @@ class GetRecordsController @Inject() (
       } yield Ok(Json.toJson(record))).merge
     }
 
+  def getRecords(
+    eori: String,
+    lastUpdatedDate: Option[String],
+    page: Option[Int],
+    size: Option[Int]
+  ): Action[AnyContent] =
+    (authAction(eori) andThen validateHeaderAction).async { implicit request =>
+      (for {
+        record <- routerService.getRecords(eori, lastUpdatedDate, page, size)
+      } yield Ok(Json.toJson(record))).merge
+    }
+
   private def validateRecordId(recordId: String): EitherT[Future, Result, String] =
     EitherT.fromEither[Future](
       Try(UUID.fromString(recordId).toString).toEither.left.map(_ =>
-        InvalidRecordIdErrorResponse(
-          dateTimeService.timestamp,
-          "Invalid record ID supplied for eori number provided"
+        InvalidErrorResponse(
+          uuidService.uuid,
```

Comment:
correlationId ID here has not meaning and it will only create confusion for the user or user support team as this is used to track the request downstream (to EIS), but in here we haven\t set the request to EIS yet